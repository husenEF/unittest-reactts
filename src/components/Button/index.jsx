const Button = (props) => {
  const { title, loading, isError } = props;

  if (loading)
    return (
      <button data-testid="btn-loading" disabled>
        loading...
      </button>
    );
  return (
    <>
      <button data-testid="btn-loading">{title}</button>
      {isError && <span>Error message</span>}
    </>
  );
};

export default Button;