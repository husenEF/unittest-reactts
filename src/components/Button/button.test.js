import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import Button from ".";

afterEach(() => {
  cleanup();
});

test("render <Button>Title</Button>", () => {
  render(<Button title="Refactory" />);
  expect(screen.getByTestId("btn-loading")).toHaveTextContent("Refactory");
});

test("render <Button title='Refactory' loading/>", () => {
  render(<Button title="Refactory" loading />);
  expect(screen.getByTestId("btn-loading")).toHaveTextContent("loading...");
  expect(screen.getByTestId("btn-loading")).toHaveProperty("disabled");
});

test("render <Button title='Refactory' isError />", () => {
  render(<Button title="Refactory" isError />);
  // render()
  expect(screen.getByText("Error message")).toBeTruthy();

  const tree = renderer.create(<Button title="Refactory" loading isError />);
  expect(tree).toMatchSnapshot();

  // expect(screen.getByTestId("btn-loading")).toHaveTextContent("Refactory");
});
