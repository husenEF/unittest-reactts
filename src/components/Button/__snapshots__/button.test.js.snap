// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <Button title='Refactory' isError /> 1`] = `
<button
  data-testid="btn-loading"
  disabled={true}
>
  loading...
</button>
`;
